digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -199,6 +226,44 @@ trait LaneOperations {
         lanePropertiesValues)
     }
 
+    // auxiliary function to find lanes which can have the correct SideCode or is BothDirection
+    def getMainLaneForBothDirection(filteredLanes: Seq[PersistedLane], laneCode: Int, sideCode: Int): Option[PersistedLane] = {
+      filteredLanes.find { lane =>
+        val isLaneCodeTowards = lane.laneCode == laneCode
+        val isSideCodeOK = lane.sideCode == sideCode || lane.sideCode == SideCode.BothDirections.value
+
+        isLaneCodeTowards && isSideCodeOK
+      }
+    }
+
+    def getLaneCodesForSideCodeAdjust(lanes: Seq[PersistedLane], laneCodeStart: Int, laneCodeEnd: Int, sideCode: SideCode ): Seq[SideCodeAdjustment] = {
+      lanes.filter { lane =>
+                    val isLaneCodeAllowed = lane.laneCode >= laneCodeStart && lane.laneCode <= laneCodeEnd
+                    val isSideCodeNOK = lane.sideCode != sideCode.value
+
+                    isLaneCodeAllowed && isSideCodeNOK
+             }
+             .map(lane => SideCodeAdjustment(lane.id, sideCode) )
+    }
+
+    /* Based on main lane number, return the range of possible lanes number.
+      ex: input: 11  || output (11,19)
+     */
+    def getLaneNumbersRange(laneNumber: Int): (Int, Int) = {
+      val laneNum = LaneNumber(laneNumber)
+
+      if (laneNumber == laneNum.againstDirection)
+        (laneNum.againstDirection, FourthRightAdditional.againstDirection)
+      else if (laneNumber == laneNum.towardsDirection)
+        (laneNum.towardsDirection, FourthRightAdditional.towardsDirection)
+      else
+        (laneNum.towardsDirection, FourthRightAdditional.towardsDirection)
+    }
```

Comment:
In the "else if" and "else" condition the result is the same, if this is correct we can do only like this:
```Scala
    def getLaneNumbersRange(laneNumber: Int): (Int, Int) = {
      val laneNum = LaneNumber(laneNumber)

      if (laneNumber == laneNum.againstDirection)
        (laneNum.againstDirection, FourthRightAdditional.againstDirection)
      else
        (laneNum.towardsDirection, FourthRightAdditional.towardsDirection)
    }

Reply:
Done.